src/cbang/openssl/SSLContext.cpp

```diff
@@ -94,6 +98,34 @@ namespace {
 }
 
 
+#ifdef __APPLE__
+// FIXME this should be moved to MacOSUtilities
+namespace {
+  const string getStringFromCFString(const CFStringRef cfstr) {
+      if (!cfstr)
+          return string();
+      // CFStringGetCString requires an 8-bit encoding
+      CFStringEncoding encoding = kCFStringEncodingUTF8;
+      const char *str0 = CFStringGetCStringPtr(cfstr, encoding);
+      if (str0)
+          return string(str0);
+      CFIndex len = 1 +
+        CFStringGetMaximumSizeForEncoding(CFStringGetLength(cfstr), encoding);
+      string myString;
+      char *buf = new char[len];
+      if (buf) {
+          if (CFStringGetCString(cfstr, buf, len, encoding))
+              myString = buf;
+          delete[] buf;
+      }
+      return myString;
+  }
+}
+#endif
+
+
+namespace cb { // avoid namespace collision with Apple Security.framework
```

Comment:
This namespacing does not seem necessary.  The ``SSLContext`` class is already in the ``cb`` namespace.

Reply:
The namespace *is* necessary for osx. Otherwise, clang reports ambiguous SSLContext for every method, because an SSLContext exists in Security.framework.

If you like, I could put the namespacing in an ifdef __APPLE__.

Reply:
I put them in an ifdef.
Absolutely needed to compile on osx.

Reply:
OK.  That makes sense.